olp-cpp-sdk-core/src/cache/DefaultCacheImpl.h

```diff
@@ -113,19 +119,37 @@ class DefaultCacheImpl {
 
   DefaultCache::StorageOpenResult SetupStorage();
 
+  bool IsProtected(const std::string& key) const;
+
   bool GetFromDiskCache(const std::string& key,
                         KeyValueCache::ValueTypePtr& value, time_t& expiry);
 
   boost::optional<std::pair<std::string, time_t>> GetFromDiscCache(
       const std::string& key);
 
+  // custom comparator needed to reduce duplicates for keys, which are already
+  // protected by prefix
+  struct CustomCompare {
+    bool operator()(const std::string& lhs, const std::string& rhs) const {
+      if (rhs.length() < lhs.length()) {
+        if (lhs.substr(0, rhs.length()) == rhs)
```

Comment:
substr() creates a copy of a string which is very inefficient. Use std::string::find().